src/prompt/variable.rs

```diff
@@ -156,6 +156,11 @@ fn quoted_str_parser<'a>(input: &'a str, quote: char) -> IResult<&'a str, String
 
                         for (i, c) in str.char_indices() {
                             if prev == None {
+                                // first iteration, we want to make sure that this is not a quote
```

Comment:
I think this works but it has gotten very tangled. (Two breaks, two continues, plus the normal loop termination condition.)

I'd recommend trying out my suggestion from the earlier comment, which I think gets it down to one break and no continues